client/tx/tx.go

```diff
@@ -38,7 +55,7 @@ func GenerateOrBroadcastTxWithFactory(clientCtx client.Context, txf Factory, msg
 	// Right now, we're factorizing that call inside this function.
 	// ref: https://github.com/cosmos/cosmos-sdk/pull/9236#discussion_r623803504
 	for _, msg := range msgs {
-		if err := msg.ValidateBasic(); err != nil {
+		if err := ValidateMsg(msg); err != nil {
```

Comment:
I don't think putting ValidateMsg here is enough. Here we're doing client-side validation. We should also do server-side validation.

IMO we should add the same logic inside the ValidateBasicDecorator ante-handler. This way, the same validation will also run when clients send a tx via gRPC or via REST.